platform-hub-api/app/controllers/kubernetes/clusters_controller.rb

```diff
@@ -1,52 +1,69 @@
 class Kubernetes::ClustersController < ApiJsonController
 
-  before_action :load_kubernetes_clusters_hash_record
+  before_action :find_cluster, only: [ :show, :update ]
+
+  skip_authorization_check only: [ :index, :show ]
+  load_and_authorize_resource class: KubernetesCluster, :except => [:index, :show]
 
   # GET /kubernetes/clusters
   def index
-    authorize! :read, :kubernetes_clusters
-    clusters = @kubernetes_clusters.data.map do |c|
-      c.with_indifferent_access.slice(:id, :description)
-    end.sort_by! do |c|
-      c[:id]
-    end
-    render json: clusters
+    clusters = KubernetesCluster.all.sort_by(&:name)
+    render json: clusters, each_serializer: KubernetesClusterSerializer
   end
 
-  # PATCH/PUT /kubernetes/clusters/:id
-  def create_or_update
-    authorize! :manage, :kubernetes_clusters
+  # GET /kubernetes/clusters/:id
+  def show
+    render json: @cluster
+  end
 
-    data = cluster_params.to_h
-    data[:id] = params[:id]  # ID in URL takes precedence
+  # POST /kubernetes/clusters
+  def create
+    cluster = KubernetesCluster.new(cluster_params)
 
-    Kubernetes::ClusterService.create_or_update data
+    if cluster.save
+      AuditService.log(
+        context: audit_context,
+        action: 'create',
+        auditable: cluster,
+        comment: "Kubernetes cluster '#{cluster.name}' created or updated by #{current_user.email}"
```

Comment:
Generally, for simple CRUD actions like these, we don't add a comment, as it doesn't add anything extra that isn't already there in the `Audit` attributes (and also saves a little extra space in DB).

If you do leave the comment in, you should remove the "or updated".